sympy/polys/polyroots.py

```diff
@@ -845,6 +845,7 @@ def roots(f, *gens, **flags):
 
     """
     from sympy.polys.polytools import to_rational_coeffs
+    from sympy.solvers.solvers import solve
```

Comment:
`solve` is a high-level routine that should not be called from this low level routine. The diff I gave (and now slightly momdified) will handle the case when a generator is not specified.